badge_server/main.py

```diff
@@ -75,160 +88,211 @@ def _get_pair_status_for_packages(pkg_sets):
+                    'detail': ('google-cloud-bigquery is 3 or more minor'
+                              'versions behind the latest version')
+                },
+            },
         }
-    }
-    for pkg_set in pkg_sets:
-        pkgs = [package_module.Package(pkg) for pkg in pkg_set]
-        pair_res = badge_utils.store.get_pair_compatibility(pkgs)
-
-        for res in pair_res:
-            py_version = badge_utils.PY_VER_MAPPING[res.python_major_version]
-            # Status showing one of the check failures
-            if res.status.value != 'SUCCESS':
-                # Ignore the package that not support for given py_ver
-                if pkg_set[1] in \
-                        configs.PKG_PY_VERSION_NOT_SUPPORTED.get(
-                            res.python_major_version):
-                    continue
-                # Ignore the package that are not self compatible
-                self_status = _get_result_from_cache(
-                    package_name=pkg_set[1],
-                    badge_type=badge_utils.BadgeType.SELF_COMP_BADGE)
-                if self_status[py_version]['status'] != 'SUCCESS':
-                    continue
-                version_and_res[py_version]['status'] = res.status.value
-                version_and_res[py_version]['details'][pkg_set[1]] = \
-                    res.details if res.details is not None \
-                    else badge_utils.EMPTY_DETAILS
-    return version_and_res
-
-
-def _get_all_results_from_cache(package_name, commit_number=None):
-    """Get all the check results from cache.
+    """
+    result_dict = badge_utils._build_default_result(
+        status='UP_TO_DATE', include_pyversion=False, details={})
+
+    outdated_deps = badge_utils.highlighter.check_package(package_name)
+    _deps_list = badge_utils.finder.get_deprecated_dep(package_name)[1]
+    deprecated_deps = ', '.join(_deps_list)
+
+    outdated_depencdency_name_to_details = {}
+    max_level = badge_utils.priority_level.UP_TO_DATE
+    for dep in outdated_deps:
+        dep_detail = {}
+        level = dep.priority.level
+        if level.value > max_level.value:
+            max_level = level
+        dep_detail['installed_version'] = dep.installed_version
+        dep_detail['latest_version'] = dep.latest_version
+        dep_detail['priority'] = dep.priority.level.name
+        dep_detail['detail'] = dep.priority.details
+        outdated_depencdency_name_to_details[dep.name] = dep_detail
+    result_dict['status'] = max_level.name
+    result_dict['details'] = outdated_depencdency_name_to_details
+    result_dict['deprecated_deps'] = deprecated_deps
+
+    return result_dict
+
+
+def _get_badge_status(
+        self_compat_res: dict,
+        google_compat_res: dict,
+        dependency_res: dict) -> str:
+    """Get the badge status.
+
+    The badge status will determine the right hand text and the color of
+    the badge.
+
+    See badge_utils.STATUS_COLOR_MAPPING.
```

Comment:
And this will match the PRD in a follow-up PR?

Reply:
yep